### ISSUE TITLE
Pattern matcher regression

### ISSUE THREAD

**Issue by emste** [explosion/spaCy#1503]
Hey!
I'm working with the Matcher and experienced changed behaviour between v2.0.0a17 and v2.0.0a19. The pattern operator '*' seems to be less greedy than it was in a17:

```
import spacy

from spacy.matcher import Matcher

nlp = spacy.load('en_core_web_sm')

def on_match(matcher, doc, i, matches):
    match_id, start, end = matches[i]
    match = doc[start : end].text
    print(f"on_match: {match}")

matcher = Matcher(nlp.vocab)
matcher.add('MyMatcher', on_match,  [ {'ORTH': 'Hello'}, {'ORTH': 'World', 'OP': '*'} ])

doc1 = nlp('Hello World')
doc2 = nlp('Hello World World World')
matcher(doc1)
matcher(doc2)
# Output for a17
# on_match: Hello World
# on_match: Hello World World World
# Output for a19
# on_match: Hello World
# on_match: Hello World
```

Is this an intended change?

## Env a17
* **spaCy version:** 2.0.0a17
* **Platform:** Linux-4.9.49-moby-x86_64-with-debian-8.9
* **Python version:** 3.6.3
* **Models:** en_core_web_sm, xx_ent_wiki_sm

## Env a19
* **spaCy version:** 2.0.0a19
* **Platform:** Linux-4.9.49-moby-x86_64-with-debian-8.9
* **Python version:** 3.6.3
* **Models:** en_core_web_sm, xx_ent_wiki_sm, de_core_news_sm

**Comment by wpm** [comment]
Looks like [the same thing](https://support.prodi.gy/t/op-does-not-appear-to-be-greedy/181) is happening with the `+` qualifier.

**Comment by ines** [comment]
Should be fixed in #1876!
